crates/telio-nurse/src/aggregator.rs

```diff
@@ -277,12 +279,18 @@ impl<W: WireGuard> ConnectivityDataAggregator<W> {
             _ => None,
         };
 
-        if event.peer_state != PeerState::Connected {
-            if let Entry::Occupied(entry) = event_entry {
-                entry.remove();
+        match event_entry {
+            Entry::Occupied(mut entry) => {
+                if event.peer_state != PeerState::Connected {
+                    entry.remove();
+                } else if update_entry {
```

Comment:
A quick question, while reading, I was not fully able to understand why "update_entry" flag is needed. Could you explain it? :pray:

Reply:
So we have 3 cases.
1) When there is already an entry, but now peer isn't connected so we remove it.
2) When there is already an entry, but now the connection state has changed so we need to overwrite the existing entry. The problem why `update_entry` was needed was because if there was already an entry, and peer state was still connected then the entry wasn't being updated.
3) When there is no entry then make one.

Reply:
@Hasan6979 - well, that was intended, when the state in the event is the same as the current state, there is no need to start another segment

Reply:
Yes. I agree. That is where the `update_entry` comes into play. When states are different, and there was already an entry in place.

Reply:
Uhhh, that's right, that was a mistake - but I guess using an additional mutable boolean flag isn't that nice, could you just use `new_segment.is_come()` instead?